src/RulesTest/PropertyConditions/ElementGroupsTests.cs

```diff
@@ -280,5 +280,54 @@ public void IsButtonText_MatchExpected()
                 Assert.IsFalse(ElementGroups.IsButtonText.Matches(e));
             } // using
         }
+
+        [TestMethod]
+        public void IsCheckBoxText_MatchExpected()
+        {
+            using (var e = new MockA11yElement())
+            using (var parent = new MockA11yElement())
+            {
+                Assert.IsFalse(ElementGroups.IsCheckBoxText.Matches(e));
+
+                e.ControlTypeId = Button;
+                Assert.IsFalse(ElementGroups.IsCheckBoxText.Matches(e));
+
+                e.ControlTypeId = Text;
+                Assert.IsFalse(ElementGroups.IsCheckBoxText.Matches(e));
+
+                e.Parent = parent;
+                parent.ControlTypeId = CheckBox;
+                Assert.IsTrue(ElementGroups.IsCheckBoxText.Matches(e));
+
+                parent.ControlTypeId = Window;
```

Comment:
I think we need to add a check for another type of child element under a checkbox parent -- that seems like a case that isn't covered. I'm also wondering if the change from Checkbox to Window isn't redundant with the checks above when e has no parent at all.

Reply:
OK, I've reduced this to 4 cases: Element is Button or Window, parent is Null or CheckBox.